### PR TITLE
Feature/more warnings

### PR DESCRIPTION
**Pull request by lingtaolf**
Original issue:  #56346

### Changelog category (leave one):
- New Feature

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add more warnings on the number of databases, tables.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by hanfei1991** [comment]
We can add some docs for the new settings

**Comment by hanfei1991** [comment]
@lingtaolf for this PR it's enough. There are also some style check failure. please fix it

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 1]
```diff
@@ -472,6 +472,39 @@ The value 0 means that you can delete all tables without any restrictions.
...
+Default value: 1000
+
+**Example**
+
+``` xml
+<max_database_num_to_warn>50</max_database_num_to_warn>
+```
+  
+## max\_table\_num\_to\_warn {#max-table-num-to-warn}   
+If the number of attached tables exceeds the specified value, clickhouse server will add warning message to `system.warnings` table.
```
Comment (hanfei1991) [inline docs/en/operations/server-configuration-parameters/settings.md:488]: warning messages

File: docs/en/operations/server-configuration-parameters/settings.md [hunk 2]
```diff
@@ -472,6 +472,39 @@ The value 0 means that you can delete all tables without any restrictions.
...
+
+**Example**
+
+``` xml
+<max_table_num_to_warn>400</max_table_num_to_warn>
+```
+
+
+## max\_part\_num\_to\_warn {#max-part-num-to-warn}  
+If the number of active parts exceeds the specified value, clickhouse server will add warning message to `system.warnings` table.
```
Comment (hanfei1991) [inline docs/en/operations/server-configuration-parameters/settings.md:499]: warning messages

File: src/Interpreters/Context.cpp [hunk 3]
```diff
@@ -323,6 +326,10 @@ struct ContextSharedPart : boost::noncopyable
     std::optional<MergeTreeSettings> replicated_merge_tree_settings TSA_GUARDED_BY(mutex);   /// Settings of ReplicatedMergeTree* engines.
     std::atomic_size_t max_table_size_to_drop = 50000000000lu; /// Protects MergeTree tables from accidental DROP (50GB by default)
     std::atomic_size_t max_partition_size_to_drop = 50000000000lu; /// Protects MergeTree partitions from accidental DROP (50GB by default)
+                                                                   ///     std::atomic_size_t max_table_size_to_warn;
```
Comment (hanfei1991) [inline src/Interpreters/Context.cpp:329]: remove this comment?

File: tests/config/config.d/max_num_to_warn.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<clickhouse>
```
[the remark this answers is not in the document]
Comment (lingtaolf) [reply tests/config/config.d/max_num_to_warn.xml:1]: Clickhouse server does not verify that the parameters are in the` ServerSettings.h` file. So I don't think it'll interfere with the other tests.

**Comment by lingtaolf** [comment]
> @lingtaolf for this PR it's enough. There are also some style check failure. please fix it

I've check the log and code and still don't know what style is wrong.

**Comment by lingtaolf** [comment]
@hanfei1991 Hi, here are some tests failed, but these seem to have nothing to do with my changes. Could you please take a look.

**Review by hanfei1991 — APPROVED** [review]

**Pull request merged**